### PR TITLE
[FX-32] Return the balance object rather than payment method on balance check

### PR DESCRIPTION
**Pull request by dleis612**
# Description
Rather than returning the entire payment method object on a successful balance check, return the balance.

### CODE REVIEW

File: forage-android/src/main/java/com/joinforage/forage/android/network/PaymentMethodService.kt [hunk 1]
```diff
@@ -22,7 +22,7 @@ internal class PaymentMethodService(
 
         val request: Request = Request.Builder()
             .url(url)
-            .header(ForageConstants.Headers.API_VERSION, "2023-03-31")
+            .header(ForageConstants.Headers.API_VERSION, "2023-05-15")
```
Comment (djoksimo) [inline forage-android/src/main/java/com/joinforage/forage/android/network/PaymentMethodService.kt:25]: can this change be in a separate PR or does this change depend on v 2023-05-15?

File: forage-android/src/main/java/com/joinforage/forage/android/network/TokenizeCardService.kt [hunk 2]
```diff
@@ -33,7 +33,7 @@ internal class TokenizeCardService(
 
         val request: Request = Request.Builder()
             .url(url)
-            .header(ForageConstants.Headers.API_VERSION, "2023-03-31")
+            .header(ForageConstants.Headers.API_VERSION, "2023-05-15")
```
Comment (djoksimo) [inline forage-android/src/main/java/com/joinforage/forage/android/network/TokenizeCardService.kt:36]: cc comment about separating the header change to a separate PR

File: sample-app/src/main/java/com/joinforage/android/example/network/ForageApi.kt [hunk 3]
```diff
@@ -10,6 +10,6 @@ import retrofit2.http.POST
 
 interface ForageApi {
     @POST("api/payments/")
-    @Headers("API-VERSION: 2023-03-31")
+    @Headers("API-VERSION: 2023-05-15")
```
Comment (djoksimo) [inline sample-app/src/main/java/com/joinforage/android/example/network/ForageApi.kt:13]: cc comment about separating api-version header to a diff pr

Comment (dleis612) [reply sample-app/src/main/java/com/joinforage/android/example/network/ForageApi.kt]: Just changed back, added those to test.

**Review by djoksimo — APPROVED** [review]
LGTM once CI checks / tests pass 🚀
